### ISSUE TITLE
Request: Stop Grav updates from wiping out 'images' folder

### ISSUE THREAD

**Issue by ghost** [getgrav/grav#3416]
I have multiple image gallery pages on my site, some with well over 100 images, with thumbnails generated using `cropZoom`. Every time I update Grav, it wipes out the `images` folder where the thumbnails are, and next time I load those pages it takes a very long time for all those thumbnails to be regenerated.

I'd like to see an option to set Grav so that it doesn't clear the `images` folder unless I explicitly tell it to. Perhaps be able to specify a permanent folder for the thumbnails to be stored in?

**Comment by rhukster** [comment]
something like this?

https://github.com/getgrav/grav/blob/b3af6c992011734b5f8ddd6480de595c0d0a3e74/system/config/system.yaml#L99

**Comment by gitanno** [comment]
I've gone and set this to `false`, I'll report back next time there's a Grav update.

**Comment by pamtbaau** [comment]
@rhukster, setting `system.cache.clear_images_by_default` is being used in `Grav/Common/Cache::clearCache()`, but only when `clearCache()` is being called with parameter `$remove` having value `default`.

Command `$ bin/grav cache` works as expected when setting `system.cache.clear_images_by_default` to `true` or `false`.

However, when Grav is being updated, `Grav/Installer/Install::finalize()` calls `Cache::clearCache('all');`. Hence the images folder is being wiped out no matter the setting for `system.cache.clear_images_by_default`

**Comment by dallaslu** [comment]
So before each update, I will back up the `images` folder.

**Comment by gitanno** [comment]
Just updated my site to 1.7.9 through the Admin Panel. The ``images`` folder still seems to have been wiped, resulting in extremely long load times on my gallery pages.

**Comment by rhukster** [comment]
Updates do a full a full cache clear.

**Comment by mahagr** [comment]
I tend to agree here, I cannot remember the last time when Grav update has changed existing images. Images are a special case of cache -- I would exclude them.

**Comment by pamtbaau** [comment]
I wonder, due to which changes in Grav do existing folder structures and images in */images* become unusable and must therefor be wiped?

- Is there a change in Grav for which the folder structure need to be changed?
- I can't think of a reason a browser can no longer load existing images from */images* (unless the folder structure changes)
- A change in image library makes sense in order to (optionally?) re-optimise cached images, but otherwise...

**Comment by rhukster** [comment]
Fixed for Grav 2.1.0. Updating Grav ran a full cache clear that ignored `cache.clear_images_by_default`, so the processed-image folder was wiped every time. Resized images now survive every cache clear and update unless that setting is on. `bin/grav cache --images-only` still removes them on demand when you want to.
